core/CronArchive/QueueConsumer.php

```diff
@@ -201,6 +201,11 @@ public function getNextArchivesToProcess()
                 break;
             }
 
+            //check if isSegmentAvailable
+            if (!$this->isSegmentAvailable($invalidatedArchive) && !Manager::getInstance()->isPluginActivated($invalidatedArchive['plugin'])) {
```

Comment:
The plugin check shouldn't be needed here as it's done a couple of lines below. Actually it might even fail if the plugin is empty.